### PR TITLE
Update for evrythng.js v5, to replace separate MQTT and WS SDKs

### PR DESCRIPTION
**Pull request by C-D-Lewis**
Continuing on from `v1.x` branch:

_Features_
- Supports `evrythng.js` v5.1.0

_Breaking Changes_
- `subscribe()`, `unsubscribe()`, and `publish()` all return Promises instead of callbacks.

### CODE REVIEW

File: src/getClient.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  client.on('connect', initClient(client, scope, resolve))
+  client.on('close', cleanUp(client, scope, reject))
+  client.on('error', cleanUp(client, scope, reject))
+  client.on('message', (path, message) => onMessage(scope, path, message))
+})
+
+/**
+ * Get a client for the given scope.
+ *
+ * @param {object} scope - evruthng.js scope.
```
Comment (adarrra) [inline src/getClient.js]: typo

File: src/getClient.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    return
+  }
+
+  delete scope.pubsubClient
+}
+
+/**
+ * When a message arrives from mqtt.js, notify all relevant subscriptions.
+ *
+ * @param {object} scope - evruthng.js scope.
```
Comment (adarrra) [inline src/getClient.js]: typo in evrythng

File: src/getClient.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  const handlers = scopeSubscriptions && scopeSubscriptions[path]
+  if (handlers) {
+    handlers.forEach(onMessage => onMessage(message))
+  }
+}
+
+/**
+ * Create a mqtt.js client object and setup connection handlers.
+ *
+ * @param {object} scope - evruthng.js scope.
```
Comment (adarrra) [inline src/getClient.js]: typo

File: src/publish.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const getClient = require('./getClient')
+
+/**
+ * Publish a message to a topic using the given client.
+ *
+ * @param {object} client - The client to publish with.
+ * @param {string} topic - The topic to publish on.
+ * @param {*} payload - String or object payload, coverted as required.
```
Comment (adarrra) [inline src/publish.js]: coNverted ?

File: tests/unit/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+      done()
+    }
+
+    trustedApp.action('all').subscribe(onMessage)
+      .then(() => trustedApp.action('all').publish({ type: 'scans', thng: thng.id }))
+  })
+
+  it('should unsubscribe from all actions', async () => {
+    await trustedApp.action('all').unsubscribe()
```
Comment (adarrra) [inline tests/unit/index.spec.js]: is this test useful?

Comment (C-D-Lewis) [reply tests/unit/index.spec.js]: It makes sure it doesn't throw, if anything.

File: src/publish.js (hunk 4 above)
Comment (C-D-Lewis) [reply src/publish.js]: Fixed

File: src/getClient.js (hunk 3 above)
Comment (C-D-Lewis) [reply src/getClient.js]: Fixed

File: src/getClient.js (hunk 2 above)
Comment (C-D-Lewis) [reply src/getClient.js]: Fixed

File: src/getClient.js (hunk 1 above)
Comment (C-D-Lewis) [reply src/getClient.js]: Fixed

**Pull request merged**